src/test/modelinstance_test.cpp

```diff
@@ -360,6 +369,108 @@ TEST_F(TestLoadModel, CheckIfNonExistingBinFileReturnsFileInvalid) {
     EXPECT_EQ(status, ovms::StatusCode::FILE_INVALID) << status.string();
 }
 
+TEST_F(TestLoadModel, CheckSavedModelHandling) {
+    ovms::ModelInstance modelInstance("saved-model", UNUSED_MODEL_VERSION, *ieCore);
+
+    const std::string modelPath = directoryPath + "/test_saved_model";
+    std::filesystem::create_directories(modelPath);
+    ovms::model_version_t version = 1;
+    const std::string versionDirectoryPath = modelPath + "/" + std::to_string(version);
+    if (!std::filesystem::exists(versionDirectoryPath)) {
+        ASSERT_TRUE(std::filesystem::create_directories(versionDirectoryPath));
+    }
+    {
+        std::ofstream savedModelFile{versionDirectoryPath + "/saved_model.pb"};
+        savedModelFile << "NOT_NEEDED_CONTENT" << std::endl;
+    }
+    const ovms::ModelConfig config{
+        "saved-model",
+        modelPath,  // base path
+        "CPU",      // target device
+        "1",        // batchsize
+        1,          // NIREQ
+        false,      // is stateful
+        false,      // idle sequence cleanup enabled
+        false,      // low latency transformation enabled
+        500,        // stateful sequence max number,
+        "",         // cache dir
+        version,    // version
+        modelPath,  // local path
+    };
+    auto status = modelInstance.loadModel(config);
+    auto model_files = modelInstance.getModelFiles();
+
+    EXPECT_EQ(model_files.front(), directoryPath + "/test_saved_model/1/");
+}
+
+TEST_F(TestLoadModel, CheckTFModelHandling) {
+    ovms::ModelInstance modelInstance("tf", UNUSED_MODEL_VERSION, *ieCore);
+
+    const std::string modelPath = directoryPath + "/test_tf";
+    std::filesystem::create_directories(modelPath);
+    ovms::model_version_t version = 1;
+    const std::string versionDirectoryPath = modelPath + "/" + std::to_string(version);
+    if (!std::filesystem::exists(versionDirectoryPath)) {
+        ASSERT_TRUE(std::filesystem::create_directories(versionDirectoryPath));
+    }
+    {
+        std::ofstream savedModelFile{versionDirectoryPath + "/model.pb"};
+        savedModelFile << "NOT_NEEDED_CONTENT" << std::endl;
+    }
+    const ovms::ModelConfig config{
+        "saved-model",
+        modelPath,  // base path
+        "CPU",      // target device
+        "1",        // batchsize
+        1,          // NIREQ
+        false,      // is stateful
+        false,      // idle sequence cleanup enabled
+        false,      // low latency transformation enabled
+        500,        // stateful sequence max number,
+        "",         // cache dir
+        version,    // version
+        modelPath,  // local path
+    };
+    auto status = modelInstance.loadModel(config);
+    auto model_files = modelInstance.getModelFiles();
```

Comment:
Assert number of model files (should be 1?). We can avoid segfaults in lin 437/403/471 in case size is 0